keps/0004-add-testing-infrastructure.md

```diff
@@ -68,21 +80,48 @@ This is a two phase proposal.
 
 ### Tests
 
-Add testing packages to:
+KUDO will be tested by several different test suites that validate individual components, KUDO, and frameworks.
+
+#### Unit tests
+
+Unit tests test a single component and require no network connectivity. These should be fast, run on every pull request,
+and test that individual KUDO components behave correctly. Unit tests are written as standard Go tests.
+
+All packages should incorporate unit tests that validate their behavior. Some examples of tests that should be included are:
 
 - pkg/apis - Conformance of objects, serialization, validation
 - pkg/controller - Controllers implement the correct process flows for changes in Instance objects, and creation of PlanExecutions
 - pkg/util - Util functions behave as expected
 
-Integration tests
+Go coverage reports will be used to ensure that changes do not reduce test coverage.
+
+#### Integration tests
+
+Integration tests test the KUDO controller in its entirety. These tests can be run against either a real Kubernetes cluster
+or a local control plane. Initially, all integration tests will be run against every pull request. If this begins to degrade
+velocity, integration tests can be run on master or nightly builds. Integration tests can be written either in Go or using
+the test harness designed in KEP-0008.
+
+The integration tests will consist of a set of representative Frameworks and FrameworkVersions that utilize real world
```

Comment:
Maybe out of scope of this, but when is the right time to add a new FrameworkVersion that tests a capability?

Reply:
Good question. I could document that it's up to the discretion of the reviewers / designers / author - or we could try to come up with something more formal like if a capability is defined in a KEP it should have a test?